### PR TITLE
Implement non-zero dimension attribute parsing

### LINKED ISSUE #8445: Dimension properties being parsed incorrectly
Continued from [this issue](https://github.com/servo/servo/issues/8417),

For example, [this section in the spec](https://html.spec.whatwg.org/multipage/rendering.html#tables-2:the-table-element-6) says that the height attribute be parsed as a **non-zero** dimension property. We are currently parsing using `util::str::parse_length`, which allows zero values. We should create a new function like `util::str::parse_length_nonzero` and use it where appropriate.

### PR DESCRIPTION
**Pull request by frewsxcv**
Fixes #8445

The only attributes I found that we have implemented that uses non-zero
dimenion attributes:
- `width` for `<td>` and `<th>` (table cells)
- `width` for `<table>`

I updated these implementations to use the new non-zero dimension
attribute parsing and added associated regression tests.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9054)

### CODE REVIEW

File: tests/wpt/web-platform-tests/html/rendering/non-replaced-elements/tables/table-width.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<link rel="match" href="table-width-ref.html">
```
Comment (frewsxcv) [inline tests/wpt/web-platform-tests/html/rendering/non-replaced-elements/tables/table-width.html]: This currently doesn't pass in Servo (for presumably unrelated reasons?) but it passes in Firefox and Chrome. If anyone has any insight into this or has a suggestion for a better test case for `width=0`, please let me know

**Comment by ecoal95** [comment]
Fails tidy:

```
./components/util/str.rs:7: use statement is not in alphabetical order
	expected: euclid::num::Zero
	found: cssparser::{self, Color, RGBA}
```

**Comment by frewsxcv** [comment]
> Fails tidy:

Thanks, fixed in the latest force push

File: tests/wpt/web-platform-tests/html/rendering/non-replaced-elements/tables/table-width.html (hunk 1 above)
Comment (eefriedman) [inline tests/wpt/web-platform-tests/html/rendering/non-replaced-elements/tables/table-width.html]: Yes, failing for unrelated reasons; as a workaround, you could try "border-spacing: 0;".

Comment (eefriedman) [inline tests/wpt/web-platform-tests/html/rendering/non-replaced-elements/tables/table-width.html]: Err, wait, nevermind, that's not the primary issue, is it?  The problem is the interaction between the width specification on table and td.

Here's a testcase which should work in servo:
```html
<table width=0>
<tr>
<td>
a b
</td>
</tr>
</table>
```

Comment (frewsxcv) [inline tests/wpt/web-platform-tests/html/rendering/non-replaced-elements/tables/table-width.html]: Thanks for the suggestion, that's a lot cleaner! Updated in the latest force push

**Comment by eefriedman** [comment]
Reviewed 10 of 14 files at r1, 1 of 1 files at r2, 2 of 3 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9054)*

**Pull request merged**